app/views/admin/users/index/_controls.html.erb

```diff
@@ -1,15 +1,54 @@
+<%= javascript_packs_with_chunks_tag "admin/members/memberControls", defer: true %>
+
 <% if FeatureFlag.enabled?(:member_index_view) %>
   <% if params[:controller] == "admin/users" %>
-    <%= form_with url: admin_users_path, method: :get, local: true, class: "flex flex-col m:flex-row gap-3 m:items-center p-3" do |f| %>
-      <div class="crayons-field flex-1 flex-row items-center gap-2">
-        <%= f.text_field :search, value: params[:search], class: "crayons-textfield mt-0", placeholder: "Search member...", aria: { label: "Search member by name, username, email, or Twitter/GitHub usernames" } %>
+    <!-- Small screen layout -->
+    <%= form_with url: admin_users_path, method: :get, local: true, class: "l:hidden" do |f| %>
+      <div class="flex justify-between my-3">
+        <div>
+          <button type="button" id="expand-search-btn" class="relative c-btn c-btn--icon-alone indicator-btn" aria-label="Expand search" aria-expanded="false" aria-controls="search-users">
+            <%= crayons_icon_tag("search", aria_hidden: true) %>
+            <span class="search-indicator absolute top-1 right-1 c-indicator c-indicator--info <%= params[:search].blank? ? "hidden" : "" %>"></span>
+          </button>
+          <button type="button" id="expand-filter-btn" class="c-btn c-btn--icon-alone indicator-btn" aria-label="Expand filter" aria-expanded="false" aria-controls="filter-users">
+            <%= crayons_icon_tag("filter", aria_hidden: true) %>
+            <span class="search-indicator absolute top-1 right-1 c-indicator c-indicator--info <%= params[:role].blank? ? "hidden" : "" %>"></span>
+          </button>
+        </div>
+        <%= paginate @users, theme: "admin", scope: @users, label: "Paginate users" %>
       </div>
-      <div class="crayons-field flex-row items-center gap-2">
-        <%= f.select :role, options_for_select(Role::ROLES, params[:role]), { include_blank: "All roles" }, class: "crayons-select mt-0" %>
+      <div>
+        <div id="search-users" class="hidden relative">
+          <%= f.text_field :search, value: params[:search], class: "crayons-textfield mt-0", placeholder: "Search member...", aria: { label: "Search member by name, username, email, or Twitter/GitHub usernames" } %>
+          <button type="submit" aria-label="Search" class="crayons-btn crayons-btn--ghost crayons-btn--s crayons-btn--icon-rounded absolute right-2 bottom-0 top-0 m-1">
+            <%= crayons_icon_tag(:search, aria_hidden: true) %>
```

Comment:
I noticed that theres some overlapping markup between the small screen layout. Maybe it would be a good idea to move this out into a partial? It doesn't have to happen right now, but we can see how the view evolves and the determine the re-use.

Reply:
Yeah I always find it a tough decision when to move out into a partial. Both the filter input + button and the search input + button repeat twice. I don't see any harm bringing them into their own small partials now 👍